gazebo_ros/CMakeLists.txt

```diff
@@ -37,6 +32,13 @@ add_library(gazebo_ros_node SHARED
   src/executor.cpp
   src/node.cpp
 )
+target_include_directories(gazebo_ros_node
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+    ${gazebo_dev_INCLUDE_DIRS}
+)
+
```

Comment:
This fix is similar in behavior as https://github.com/ros/console_bridge/pull/63. `gazebo_dev` exports GTest headers which are different to what `ament_cmake_gtest` provides and ultimately clashes - at least on my OSX machine.